pkgs/development/python-modules/natural/default.nix

```
@@ -0,0 +1,25 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+}:
+buildPythonPackage rec {
+  pname = "natural";
+  version = "0.2.0";
+  format = "setuptools";
```

```suggestion
  pyproject = true;
  
  build-system = [ setuptools ];
```
Please use `pyproject = true` to avoid the legacy build.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

---

when using `pyproject = true` there is an error during the build, is that a bad thing to use the legacy build ?

---

I guess you lost `six` in the `dependencies`.

`pyproject = true` let the package use `pyproject-build`. It can standardizes and simplifies Python project builds, ensuring compatibility, reducing configuration complexity, and enhancing flexibility with a centralized `pyproject.toml` configuration.

---

Sorry to ask but then how can I fix the build issue 😅 i'm not really experienced when building python packages with nix

---

Could you please provide some debug information like error log? I didn’t meet build issue with my suggestions.

---

of course, here is what I have when trying to build using `pyproject = true;`
```
ERROR Backend 'setuptools.build_meta:__legacy__' is not available.
error: builder for '/nix/store/dzcyibsbfnryhz4xr169baq8hkbw625s-python3.11-natural-0.2.0.drv' failed with exit code 1;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       > ERROR Backend 'setuptools.build_meta:__legacy__' is not available.
       For full logs, run 'nix log /nix/store/dzcyibsbfnryhz4xr169baq8hkbw625s-python3.11-natural-0.2.0.drv'.
```

---

Are you sure you added `build-system` section?

---

you were right I didn't, that fixed it, thank you 😄 but is there any documentation on how to package python other than the `python.section.md` ?

---

As far as I know, there are no beginner friendly tutorial on how to package a python module with nix. The best I could found is looking at other package for now